### PR TITLE
[REF] web_tour: totally remove allowDisabled key from step structure

### PR DESCRIPTION
**Pull request by pipu-odoo**
Until 6dbdc8c69eab2aaa6e9652a70254458fe466f791, allowDisabled key in step tour structure is no longer used. In this commit, we remove dead code.

task~3974087

https://github.com/odoo/enterprise/pull/66674

### CODE REVIEW

**Review by ltinel — APPROVED** [review]
LGTM for `sale` and `website_sale`

**Review by kmagusiak — APPROVED** [review]

File: addons/web_tour/static/src/tour_service/tour_service.js [hunk 1]
```diff
@@ -70,7 +61,6 @@ function checkTourStepKeyValues(tourStep) {
         position: { type: String, optional: true },
         run: { type: [String, Function], optional: true },
         allowInvisible: { type: Boolean, optional: true },
-        allowDisabled: { type: Boolean, optional: true },
         in_modal: { type: Boolean, optional: true },
         timeout: { type: Number, optional: true },
         consumeEvent: { type: String, optional: true },
```
Comment (BastienFafchamps) [inline addons/web_tour/static/src/tour_service/tour_service.js:66]: The `consumeEvent` key is removed from the jsdoc but is still used here, is this normal?

Comment (pipu-odoo) [reply addons/web_tour/static/src/tour_service/tour_service.js:66]: well seen !

**Comment by BastienFafchamps** [comment]
@robodoo r+

**Comment by BastienFafchamps** [comment]
@robodoo r+

**Pull request closed without merging**